src/main/java/io/tiledb/java/api/Query.java

```diff
@@ -1304,6 +1304,47 @@ public Object getCoordinates() throws TileDBError {
     return getBuffer(tiledb.tiledb_coords());
   }
 
+  /**
+   * @return The number of elements in the result NIO buffers. This is a map from the attribute name
+   *     to a pair of values.
+   *     <p>The first is number of elements for var size attributes, and the second is number of
+   *     elements in the data buffer. For fixed sized attributes (and coordinates), the first is
+   *     always 0.
+   * @param typeSize the typeSize of the attribute/dimension
+   * @exception TileDBError A TileDB exception
+   */
+  public HashMap<String, Pair<Long, Long>> resultBufferElementsNIO(int typeSize)
+      throws TileDBError {
+    HashMap<String, Pair<Long, Long>> result = new HashMap<String, Pair<Long, Long>>();
+    for (Map.Entry<String, Pair<ByteBuffer, ByteBuffer>> entry : byteBuffers_.entrySet()) {
+      String name = entry.getKey();
+
+      // Fixed-sized
+      if (entry.getValue().getFirst() == null) {
+        BigInteger val_nbytes = buffer_sizes_.get(name).getSecond().getitem(0);
```

Comment:
Should we cast this to a known max size integer so that we get an error if it is too big?

Reply:
BigInteger will throw an ArithmeticException when the result is out of the supported range. I am not sure I understood your comment, sorry..

Reply:
Added the tests and made some minor changes to the code.